presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -518,6 +520,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            List<Expression> argumentsNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
+                    .collect(toImmutableList());
+            if (!argumentsNotInGroupBy.isEmpty()) {
+                throw new SemanticException(
+                        INVALID_PROCEDURE_ARGUMENTS,
+                        node,
+                        "Arguments in GROUPING() must be in the query's GROUP BY columns. Remove columns '%s' from GROUPING() or add them to the GROUP BY.",
```

Comment:
Generally +1, but I'm not sure whether "associated query level" terminology is consistent with Presto.
Maybe `The arguments to GROUPING must be expressions referenced by the query's GROUP BY`? 

Also, the sentence `Remove columns '%s' from GROUPING() or add them to the GROUP BY.` is nice, but is it necessary? The query author will know what to do anyway. (BTW it renders with a lot of punctuation characters: `Remove columns '["a"]' from GROUPING() ...`, but that's minor)